File: frontend/app/src/interfaces/wallet.ts
export interface WalletItem {
  currency: string;
  available_balance: string;
  frozen_balance: string;
}

export interface WalletLedgerItem {
  type: string;
  amount: string;
  balance_after: string;
  ref_order_no: string | null;
  created_at: string;
}

export interface ListWalletsResponse {
  list: WalletItem[];
}

export interface ListWalletLedgersParams {
  currency: string;
  limit?: number;
  offset?: number;
}

export interface ListWalletLedgersResponse {
  list: WalletLedgerItem[];
  total: number;
}

export interface CryptoDepositInfo {
  address: string;
  memo: string;
  network: string;
  currency: string;
  contract_address: string;
}

export interface CryptoWithdrawRequest {
  to_address: string;
  amount: string;
}

export interface CryptoWithdrawResponse {
  id: number;
  status: string;
}

export interface FiatDepositRequest {
  amount: number;
}

export interface FiatDepositResponse {
  merchant_trade_no: string;
  payment_url: string;
  form_params: Record<string, string>;
}

export interface FiatWithdrawRequest {
  amount: string;
  bank_code: string;
  bank_account: string;
  account_name: string;
}

export interface FiatWithdrawResponse {
  id: number;
  status: string;
}

export interface CryptoDepositItem {
  id: number;
  currency: string;
  amount: string;
  txHash: string;
  status: 'pending' | 'confirmed' | 'failed';
  confirmedAt: string | null;
  createdAt: string;
}

export interface ListCryptoDepositsResponse {
  list: CryptoDepositItem[];
  total: number;
}

export interface CryptoWithdrawalItem {
  id: number;
  currency: string;
  amount: string;
  toAddress: string;
  txHash: string | null;
  status: 'pending' | 'broadcasting' | 'confirmed' | 'failed';
  confirmedAt: string | null;
  createdAt: string;
}

export interface ListCryptoWithdrawalsResponse {
  list: CryptoWithdrawalItem[];
  total: number;
}

File: backend/cmd/v1-p2p-exchange-service/internal/interfaces/app/wallet.go
package app_interface

// ListWalletLedgersRequest 錢包帳本流水查詢條件。
// currency 取自路徑參數，limit/offset 取自 query string。
type ListWalletLedgersRequest struct {
	Currency string `path:"currency"`
	Limit    int64  `form:"limit,optional,default=20"`
	Offset   int64  `form:"offset,optional,default=0"`
}

// WalletItem 單一幣別錢包餘額。
// 餘額欄位為字串（NUMERIC 以 ::text 承接），避免 float 精度誤差。
type WalletItem struct {
	Currency         string `json:"currency"`
	AvailableBalance string `json:"availableBalance"`
	FrozenBalance    string `json:"frozenBalance"`
}

type ListWalletsResponse struct {
	List []WalletItem `json:"list"`
}

// WalletLedgerItem 單筆錢包帳本流水。
// Amount 正數為增加、負數為減少；CreatedAt 為 UTC RFC3339。
type WalletLedgerItem struct {
	Type         string  `json:"type"`
	Amount       string  `json:"amount"`
	BalanceAfter string  `json:"balanceAfter"`
	RefOrderNo   *string `json:"refOrderNo"`
	CreatedAt    string  `json:"createdAt"`
}

type ListWalletLedgersResponse struct {
	List  []WalletLedgerItem `json:"list"`
	Total int64              `json:"total"`
}
